keras/layers/core/dense.py

```diff
@@ -94,17 +97,23 @@ def __init__(
         self.kernel_constraint = constraints.get(kernel_constraint)
         self.bias_constraint = constraints.get(bias_constraint)
         self.lora_rank = lora_rank
+        self.amax_history_length = amax_history_length
```

Comment:
I think this should not be set on the layer as a constructor arg, but on the DTypePolicy. If you want to configure it, call the layer with `layer.quantize(Float8DtypePolicy(amax_history_length=32))` or something like that. We may want to expose dedicate DTypePolicy subclasses to expose such arguments (this is the first one we encounter but there may be more of them in the future).

Reply:
I struggled to set args to DTypePolicy and I'm uncertain if this is a good approach.

In the current codebase, configuring DTypePolicy for individual layers is challenging (as shown in #19381 ).
1. DTypePolicy is treated as a global state. Making it difficult to set different `amax_history_length` for different layers. (We also need to consider scenarios involving subclasses)
2. The serialization/deserialization may be an issue.

Based on these concerns, I adopted the approach simiar to `lora_rank` that `amax_history_length` defaults to `None` and is overridden as `1024` in `self._float8_build` if not specified.

What do you think?

Reply:
I would strongly prefer to keep dtype configuration and layer constructor args separate. All dtype configuration should go through DTypePolicy.


> We also need to consider scenarios involving subclasses

If the dtype policy is a settable attribute, this is not a big issue -- you can instantiate the subclass, and then set the dtype policy on a specific layer owned by the subclass.

For serialization, the dtype policy (and its args) should be part of the layer config, under the `dtype` entry in the config dict.

`lora_rank` is simply a shortcut to 1. instantiate layer, 2. call `enable_lora()`. It's not necessary from a configuration standpoint.

Reply:
Got it. The idea of keeping the configuration minimal for layers is solid.

As we discussed above, I have introduced a `QuantizedFloat8DTypePolicy` in this PR. I think it would clearer to use "Quantized" in the naming, as it involves quantizing and dequantizing.

BTW, I'd like to point it out that setting the dtype policy after the instantiation would not trigger the quantization process.
We need to either pass the correct `dtype_policy` (such as "float8_from_float32") though `dtype` argment in `__init__` or utilize `quantize` api.

Reply:
> BTW, I'd like to point it out that setting the dtype policy after the instantiation would not trigger the quantization process.

Do you think it would be a good idea to implement a setter that checks whether the policy is a quantized policy, and calls `.quantize()` if it is?

Reply:
> Do you think it would be a good idea to implement a setter that checks whether the policy is a quantized policy, and calls `.quantize()` if it is?

This sounds like a great idea!
I think this change will also improve the UX of serializing and deserializing quantizable layers.